src/Koffee/Utility.fs

```diff
@@ -36,6 +36,40 @@ module Format =
         else if size > scale 1 then scaledStr size 1
         else scaledStr size 0
 
+    type ISubstitutionProvider =
+        abstract member getSubstitution : key:string -> string option
+
+    type EnvSubstitutionProvider() =
+        static member instance = new EnvSubstitutionProvider()
+
+        interface ISubstitutionProvider with
+            member _.getSubstitution key =
+                Environment.GetEnvironmentVariable(key, EnvironmentVariableTarget.Process)
+                |> Option.ofObj
+
+    let private subRegex = new Regex(@"%([^%]+)%", RegexOptions.Compiled)
+
+    /// Substitutes variables in string with percent syntax.
+    /// Example input: "Hello %target%!"
+    /// Outputs: "Hello world!"
+    /// Where the "target" maps to "world"
+    let subVars (subs: ISubstitutionProvider) str =
+        let mayContainSubstitution (str: string) =
+            str.IndexOf '%' <> -1
```

Comment:
Maybe want to check if there's at least 2 percentage characters.
There's a variant of `.IndexOf` that takes a starting index (<https://docs.microsoft.com/en-us/dotnet/api/system.string.indexof?view=netcore-3.1#System_String_IndexOf_System_Char_System_Int32_>). Could use that to ensure we have 2 %'s

Reply:
Skipping this. It's meant to be a quick check anyway to dismiss the most used path in this program where it does not substitute anything.